src/verilator.ts

```diff
@@ -200,17 +210,15 @@ export class VerilatorDiagnostics {
     }
 
     private _splitTerms(line: string): string[] {
-        let terms = line.split(':');
-
-        for (var i = 0; i < terms.length; i++) {
-            if (terms[i] == ' ') {
-                terms.splice(i, 1);
-                i--;
-            }
-            else {
-                terms[i] = terms[i].trim();
-            }
-        }
+        // RegExp expression for matching Verilog messages
+        // Group 1: Severity
+        // Group 2: Type (optional)
+        // Group 3: Filename (optional)
+        // Group 4: Line number
+        // Group 6: Column number (optional)
+        // Group 7: Message
+        const regex = /(Error|Warning)(-[A-Z0-9_]+)?: ([A-Za-z0-9\_\-\.]+)?:(\d+):((\d+):)? (.*)/i;
```

Comment:
Why the restrictions in file name? Verilator documentation gives the regex "%(Error|Warning)(-[A-Z0-9_]+)?: ((\S+):(\d+):((\d+):)? )?.*" (copied from https://verilator.org/guide/latest/warnings.html).

Also in order to make it easy in terms of maintenance, maybe changing the regex as below
`/(Error|Warning)(-[A-Z0-9_]+)?: ([A-Za-z0-9\_\-\.]+)?:(\d+):(?:(\d+):)? (.*)/i`
makes the groups continuous.